### PR TITLE
ci: Run functional tests on s390x

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Requested by @sipa here https://github.com/bitcoin/bitcoin/pull/17591#issuecomment-558292785

File: ci/test/00_setup_env_s390x.sh [hunk 1]
```diff
@@ -7,12 +7,10 @@
 export LC_ALL=C.UTF-8
 
 export HOST=s390x-unknown-linux-gnu
-export DOCKER_NAME_TAG=s390x/ubuntu:18.04
-export PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libssl1.0-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libqrencode-dev"
+export DOCKER_NAME_TAG=ubuntu:18.04
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_s390x.sh]: This is the default, so might as well remove it

**Comment by MarcoFalke** [comment]
Also fixed a bug in the functional tests

See https://docs.python.org/3/library/struct.html#byte-order-size-and-alignment

**Comment by practicalswift** [comment]
Concept ACK

Very happy to see this added!

And very satisfying to see that this caught a bug directly :)

**Comment by meshcollider** [comment]
Concept ACK

**Comment by laanwj** [comment]
ACK if it doesn't make Travis much slower, a bit slower may be worth it to have a full big-endian CI run.

**Comment by MarcoFalke** [comment]
For some reason this is one of the fastest machines on travis. Installing all packages only took 49 seconds (instead of minutes) and a build from scratch + unit tests + functional tests ran in 15 minutes. I'd argue that we should switch all native builds to run on that architecture instead of amd64.

**Comment by elichai** [comment]
@MarcoFalke So apparently I did all that too today lol. 
But also got the depends to work.
So up to you, if you want to cherry pick those or what exactly https://github.com/bitcoin/bitcoin/compare/master...elichai:2019-11-s390x

**Comment by MarcoFalke** [comment]
@elichai Sweet. Looks like you fixed a bug in our depends qt build. So a fresh PR might be the best.

**Pull request merged**